### PR TITLE
fix(core): Make node execution order configurable, and backward-compatible

### PR DESCRIPTION
**Pull request by netroy**

### CODE REVIEW

File: packages/editor-ui/src/components/WorkflowSettings.vue [hunk 1]
```diff
@@ -421,9 +446,14 @@ export default defineComponent({
 			saveDataSuccessExecutionOptions: [] as Array<{ key: string; value: string }>,
 			saveExecutionProgressOptions: [] as Array<{ key: string | boolean; value: string }>,
 			saveManualOptions: [] as Array<{ key: string | boolean; value: string }>,
+			executionOrderOptions: [
+				{ key: 'legacy', value: 'Legacy' },
+				{ key: 'v1', value: 'Left to Right, Top to Bottom' },
```
Comment (csuermann) [inline packages/editor-ui/src/components/WorkflowSettings.vue:451]:
```suggestion
				{ key: 'legacy', value: 'Legacy (breadth-first)' },
				{ key: 'v1', value: 'v1 (depth-first)' },
```

File: packages/editor-ui/src/stores/workflows.store.ts [hunk 2]
```diff
@@ -135,10 +140,7 @@ export const useWorkflowsStore = defineStore(STORES.WORKFLOWS, {
 			return this.workflow.versionId;
 		},
 		workflowSettings(): IWorkflowSettings {
-			if (this.workflow.settings === undefined) {
-				return {};
-			}
-			return this.workflow.settings;
+			return this.workflow.settings ?? defaults.settings;
```
Comment (janober) [inline packages/editor-ui/src/stores/workflows.store.ts:143]: If the workflow settings are empty, we return the settings object (which would be a reference). Not sure how else it is used but I assume there is the risk that if the settings of the workflow get changed, it changes by accident also the defaults for that session.

File: packages/cli/src/workflows/workflows.controller.ee.ts [hunk 3]
```diff
@@ -126,6 +126,7 @@ EEWorkflowController.post(
 		Object.assign(newWorkflow, req.body);
 
 		newWorkflow.versionId = uuid();
+		newWorkflow.settings = { ...newWorkflow.settings, executionOrder: 'v1' };
```
Comment (janober) [inline packages/cli/src/workflows/workflows.controller.ee.ts:129]: Is this necessary? Should not the front end already send it already correctly? Seems strange that if the workflow-create-request we receive would particularly mention creating it for `v0` we would overwrite it.

File: packages/cli/src/workflows/workflows.controller.ts [hunk 4]
```diff
@@ -56,6 +56,7 @@ workflowsController.post(
 		Object.assign(newWorkflow, req.body);
 
 		newWorkflow.versionId = uuid();
+		newWorkflow.settings = { ...newWorkflow.settings, executionOrder: 'v1' };
```
Comment (janober) [inline packages/cli/src/workflows/workflows.controller.ts:59]: Same as above

File: packages/cli/src/workflows/workflows.controller.ts [hunk 5]
```diff
@@ -140,7 +141,7 @@ workflowsController.get(
 			!req.user.settings?.isOnboarded &&
 			(await isBelowOnboardingThreshold(req.user));
 
-		return { name, onboardingFlowEnabled };
+		return { name, onboardingFlowEnabled, settings: { executionOrder: 'v1' } };
```
Comment (janober) [inline packages/cli/src/workflows/workflows.controller.ts:144]: Same as above

File: packages/cli/src/workflows/workflows.controller.ee.ts (hunk 3 above)
Comment (netroy) [reply packages/cli/src/workflows/workflows.controller.ee.ts]: I originally couldn't get frontend to send the payload correctly, and added this code.
but then managed to fix this in the frontend, and forgot to revert this.

**Comment by janober** [comment]
Got released with [`n8n@1.0.2`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.0.2)

**Pull request merged**